packages/common/src/store/cache/reducer.ts

```diff
@@ -41,21 +58,22 @@ export const initialCacheState = {
   // id => Set({kind, uid})
   subscriptions: {}, // things this id is subscribing to,
   // Set { id }
-  idsToPrune: new Set()
+  idsToPrune: new Set(),
+  cacheType: 'normal'
 }
 
 // Wraps a metadata into a cache entry
-const wrapEntry = (metadata: any) => ({
+const wrapEntry = (metadata: any, _timestamp?: number) => ({
   metadata,
-  _timestamp: Date.now()
+  _timestamp: _timestamp ?? Date.now()
 })
 
 // Unwraps a cache entry into its public metadata
 const unwrapEntry = (entry: { metadata: any }) => {
   if (entry && entry.metadata) {
     return entry.metadata
   }
-  return {}
+  return null
```

Comment:
This could cause runtime errors, in non ts files we could be expecting empty object

Reply:
so not having this resulting in error, since "existing" always returned true, which was not expected behavior i think based on reading through the saga. note this is just for unwrapping in the saga.